docs/graphql_design_doc.md

```diff
@@ -0,0 +1,140 @@
+# GraphQL Server – Design Document (v0.1)
+## 1. Objective
+
+The goal of this feature is to introduce a GraphQL API layer between the dashboard frontend and the existing backend services.
+This API will expose read-only experiment data (experiments --> trials --> runs --> metrics) so the dashboard can fetch exactly what it needs with a single query per view.
+This is the minimal required step to support the dashboard layout work described in:
+Issue #61 – Experiment layout in the dashboard.
+
+## 2. Scope (v0.1)
+
+We incude the following
+- A new FastAPI + Strawberry GraphQL server
+- GraphQL schema read-only
+
+- Queries (The following queries will be implemented in v0.1):
+```
+    experiments
+    experiment(uuid)
+    trials(experiment_uuid)
+    runs(trial_uuid)
+    metrics(run_uuid)
+```
+
+GraphQL resolvers mapped to existing SQLAlchemy models
+Add /graphql endpoint
+
+
+Not included （future versions）:
+Mutations, Authetication, Caching, Filtering, Pagination
+
+
+## 3. Architecture:
+Dashboard -->  GraphQL Server (FastAPI + Strawberry) --> Backend Services (SqlAlchemy/ Postgres)
+
+
+## 4. Schema Proposal (v0.1)
+### 4.1 Types
+```
+type Experiment {
```

Comment:
Let's add Projects as well. Projects is for multi-tenant, for example, for multi-teams, each one will have a different project_id.